src/App.svelte

```diff
@@ -0,0 +1,216 @@
+<script>
+  import { db } from "./firebase.js";
+  import { onMount } from "svelte";
+  import AddItemForm from "./AddItemForm.svelte";
+  import GroceryItem from "./components/GroceryItem.svelte";
+
+  let name;
+  let newItemList = {};
+
+  let groceryList = [];
+  let shareIconImage = "./images/share-icon.svg";
+  let bellIconImage = "./images/bell-regular.svg";
+  let redoIconImage = "./images/redo-solid.svg";
+  let noteIconImage = "./images/sticky-note-regular.svg";
+  let calendarIconImage = "./images/calendar.svg";
+  let dateString;
+  let isEditing = false;
+  let xxx = "";
+
+  let now = new Date(),
+    month,
+    day,
+    year;
+
+  onMount(() => {
+    db.collection("grocery").onSnapshot((snapData) => {
+      groceryList = snapData.docs;
+    });
+    (month = "" + (now.getMonth() + 1)),
+      (day = "" + now.getDate()),
+      (year = now.getFullYear());
+    if (month.length < 2) month = "0" + month;
+    if (day.length < 2) day = "0" + day;
+    dateString = [year, month, day].join("-");
+  });
+
+  function removeItem(itemId) {
+    console.log("The event works");
+    console.log(itemId);
+    db.collection("grocery").doc(itemId).delete();
+  }
+
+  // async function updateItem(itemId, newItemValue) {
+  //    const response = await db.collection("grocery").doc(itemId).update({ name: newItemList[itemId] });
+  //   newItemList[itemId] = '';
+  //   isEditing = false;
+
+  // }
+
+  function editItem() {
+    isEditing = true;
+  }
+
+  function updateItem(itemId, inputEl) {
+    console.log(itemId);
+    console.log(inputEl);
+    console.log(newItemList);
+    db.collection("grocery").doc(itemId).update({ name: inputEl });
```

Comment:
Maybe in this project is already too much but I highly recommend you to learn about asynchronous code and how to handle it properly.

This youtuber FunfunFunction has great videos about JavaScript stuff. Please, do learn and practice dealing with asyncronous code following his video tutorial:
1. What are Promises: https://www.youtube.com/watch?v=2d7s3spWAzo&list=PL0zVEGEvSaeEd9hlmCXrk5yUyqUag-n84&ab_channel=FunFunFunction
2. Async await: https://www.youtube.com/watch?v=568g8hxJJp4&t=922s&ab_channel=FunFunFunction